enhancements/cli-in-cluster-management.md

```diff
@@ -0,0 +1,219 @@
+---
+title: supported-cli-management
+authors:
+  - "@sallyom"
+reviewers:
+  - "@soltysh"
+approvers:
+  - "@soltysh"
```

Comment:
Please add @spadgett as an approver here.  

I'm not sure I fully understood what or if there would be a migration path from the existing, console shipped resource to the other and if we're going to commit to supporting both for a time since we already have a published CRD for this.  Or why we wouldn't just repurpose that resource with links pointing in cluster for anything not currently doing it.  @spadgett will need to be involved with that.

(we're currently providing in-cluster downloads for each platform's `oc` client [here](https://github.com/openshift/console-operator/blob/6960d1a90518f59f51d8836cf48d5e5595a8463c/pkg/console/controllers/clidownloads/controller.go#L150) right, Sam?)

Same for how we evolve the `cli-downloads` artifact we currently ship that supports those in-cluster downloads.

Overall I like the idea of shipping via images and providing it all in cluster to support the disconnected experience :+1:

Reply:
/cc @jhadvig - console operator maintainer :)

Reply:
This has been on the back burner for quite awhile. I'll add a migration path, but yes, will be a seamless transition, and the CRD should be repurposed. The add is that for supported CLIs (those currently offered via cli-downloads) there will be an image (within release payload? as `oc` currently is) that will allow access to binaries from mirrored images if disconnected. I'll give this enhancement an update in the coming days, thanks

Reply:
might start out as a cli plugin w/ a configurable list of cli images.  again, will give this some thought and update this proposal

Reply:
please keep in mind that we'll want a corresponding user experience via the console.  Any logic you put in the plugin either needs to be in a sharable library or API or we'll end up having to duplicate the logic in the UI.  Thanks!!